src/openrct2-ui/windows/ObjectLoadError.cpp

```diff
@@ -154,29 +154,35 @@ static rct_string_id get_object_type_string(const rct_object_entry *entry)
 *  Returns a newline-separated string listing all object names.
 *  Used for placing all names on the clipboard.
 */
-static utf8* combine_object_names(rct_window *w)
+static void copy_object_names_to_clipboard(rct_window *w)
 {
-    if (w->no_list_items > OBJECT_ENTRY_COUNT || w->no_list_items == 0) {
-        // Something's gone wrong, this shouldn't happen
-        // We don't want to allocate stupidly large amounts of memory
-        // for no reason, so bail
-        return nullptr;
-    }
-    utf8* buffer;
+    // Something has gone wrong, this shouldn't happen.
+    // We don't want to allocate stupidly large amounts of memory for no reason
+    assert(w->no_list_items > 0 && w->no_list_items <= OBJECT_ENTRY_COUNT);
 
     // No system has a newline over 2 characters
     size_t line_sep_len = strnlen(PLATFORM_NEWLINE, 2);
     size_t buffer_len = (w->no_list_items * (8 + line_sep_len)) + 1;
-    buffer = (utf8*)malloc(buffer_len);
-    buffer[0] = '\0';
+    utf8* buffer = new utf8[buffer_len]{};
```

Comment:
I haven't seen such syntax yet, what does `{}` do in this case?

Reply:
It initializes the buffer to zero.

Reply:
`utf8 * buffer`

Reply:
^ Can you fix above thingy?

Reply:
I think clang elided this whole allocation. 👍

Reply:
I mean the formatting.

Reply:
I've updated it. Also I'm still curious to which is prefered, this buffer, or `std::string`.